src/core/shape_inference/include/search_sorted_shape_inference.hpp

```diff
@@ -10,21 +10,43 @@
 namespace ov {
 namespace op {
 namespace v15 {
-template <class TShape, class TRShape = result_shape_t<TShape>>
+template <class TShape, class TRShape = result_shape_t<TShape>, class TDim = typename TShape::value_type>
 std::vector<TRShape> shape_infer(const SearchSorted* op, const std::vector<TShape>& input_shapes) {
-    // [HACK]: By convention, shape_infer should also perform node validation..
-    op->validate();
     const auto& sorted_shape = input_shapes[0];
     const auto& values_shape = input_shapes[1];
-    auto output_shape = values_shape;
-    TShape::merge_into(output_shape, sorted_shape);
-
-    if (output_shape.rank().is_static()) {
-        auto last_it = output_shape.end() - 1;
-        if (values_shape.rank().is_static()) {
-            *last_it = *(input_shapes[1].end() - 1);
-        } else {
-            *last_it = Dimension::dynamic();
+
+    if (sorted_shape.rank().is_static()) {
+        NODE_SHAPE_INFER_CHECK(op,
+                               input_shapes,
+                               sorted_shape.rank().get_length() > 0,
+                               "The sorted sequence input cannot be a scalar.");
+    }
+
+    TRShape output_shape;
+    if (values_shape.rank().is_dynamic() && sorted_shape.rank().is_static()) {
+        output_shape = sorted_shape;
+        output_shape[sorted_shape.size() - 1] = Dimension::dynamic();
+    } else {
+        output_shape = values_shape;
+    }
+
+    if (values_shape.rank().is_static() && values_shape.rank() == 0) {
+        NODE_SHAPE_INFER_CHECK(op,
+                               input_shapes,
+                               sorted_shape.rank() == 1,
+                               "If values input is a scalar the sorted sequence must be a 1D tensor.");
+    }
+
+    if (sorted_shape.rank().is_static() && values_shape.rank().is_static() && sorted_shape.rank().get_length() > 1) {
+        NODE_SHAPE_INFER_CHECK(op,
+                               input_shapes,
+                               sorted_shape.rank().compatible(values_shape.rank()),
+                               "The inputs' ranks have to be compatible.");
```

Comment:
When it is assured that both ranks are static (like here), it would be simpler to call `sorted_shape.size() == values_shape.size()`.

Reply:
Done.